src/main/java/com/adq/jenkins/xmljobtodsl/dsl/strategies/custom/DSLCleanupPatternStrategy.java

```diff
@@ -0,0 +1,93 @@
+package com.adq.jenkins.xmljobtodsl.dsl.strategies.custom;
+
+import com.adq.jenkins.xmljobtodsl.dsl.strategies.DSLObjectStrategy;
+import com.adq.jenkins.xmljobtodsl.parsers.PropertyDescriptor;
+import java.util.ArrayList;
+import java.util.List;
+
+/*
+Strategy for preBuildCleanup attribute, to correctly render includePattern and excludePattern, which is in the XML as:
+<hudson.plugins.ws__cleanup.PreBuildCleanup plugin="ws-cleanup@0.43">
+            <patterns>
+                <hudson.plugins.ws__cleanup.Pattern>
+                    <pattern>results/*</pattern>
+                    <type>INCLUDE</type>
+                </hudson.plugins.ws__cleanup.Pattern>
+            </patterns>
+            <deleteDirs>false</deleteDirs>
+            <cleanupParameter>test</cleanupParameter>
+        </hudson.plugins.ws__cleanup.PreBuildCleanup>
+
+ The value within type determines the method name: either "includePattern" or "excludePattern"
+ "pattern" is then inserted as the value for that method
+  Returns the dsl as:
+   preBuildCleanup {
+			includePattern("test")
+			deleteDirectories(false)
+			cleanupParameter("test")
+		}
+ */
+
+public class DSLCleanupPatternStrategy extends DSLObjectStrategy {
+
+    public DSLCleanupPatternStrategy(int tabs, PropertyDescriptor propertyDescriptor, String name) {
+        this(tabs, propertyDescriptor, name, true);
+    }
```

Comment:
One thing I'm curious about, if this strategy is called like this then all your work won't be used. Perhaps we should extract your code out into a function and call it in both places

Reply:
Make sure to run the extraction code in this method as well